gytheio-content-handlers/gytheio-content-handler-s3/src/main/java/org/gytheio/content/handler/s3/S3ContentReferenceHandlerImpl.java

```diff
@@ -281,15 +295,70 @@ public InputStream getInputStream(ContentReference contentReference, boolean wai
         }
     }
 
+    @Override
+    public long putFile(File sourceFile, ContentReference targetContentReference)
+            throws ContentIOException
+    {
+        if (! isContentReferenceSupported(targetContentReference))
+        {
+            throw new ContentIOException("ContentReference not supported");
+        }
+
+        long startTime = System.currentTimeMillis();
+
+        String remotePath = getRelativePath(targetContentReference.getUri());
+
+        try
+        {
+            Long contentLength = sourceFile.length();
+
+            try
+            {
+                Upload xfer = tm.upload(s3BucketName, remotePath, sourceFile);
+                xfer.waitForCompletion();
+            }
+            catch (InterruptedException e)
+            {
+                throw new ContentIOException("Failed to write content", e);
```

Comment:
I'm not sure this is how we want to handle an `InterruptedException`.

I think we want to at least set the interrupt flag and may want to consider having `putFile` (and potentially `putInputStream`) declare throwing `InterruptedException`.

See https://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java

Reply:
TODO ... I'll take a look, thanks.